main.nf

```diff
@@ -657,17 +725,25 @@ process index_chromatograms {
 }
 
 
+// Combine osw files and chromatograms by experimental design condition
+osw_for_dialignr
+ .transpose()
+ .join(chromatogram_files_indexed, by:1)
+ .groupTuple(by:0)
+ .flatMap{it -> [tuple(it[0],it[1].unique()[0],it[2].unique()[0],it[3].unique()[0],it[4],it[5],it[6])]}
```

Comment:
I think those magic indices need at least some documentation to what column they refer to

Reply:
yes, if it comes from the csvreader, you can e.g. use named columns I think.

Reply:
I made a comment, noting what each index stands for in the channel after the transposing, joining and grouping.